backend/tests/unit/services/test_stage_execution_manager.py

```
@@ -0,0 +1,471 @@
+        history_service = Mock()
+        history_service.is_enabled = True
+        history_service.get_repository = Mock(return_value=Mock(__enter__=Mock(return_value=Mock(session=Mock()))))
+        history_service._retry_database_operation_async = AsyncMock(return_value=True)
+        
+        manager = StageExecutionManager(history_service=history_service)
+        
+        stage = SimpleNamespace(name="test-stage", agent="TestAgent")
+        
+        with patch('tarsy.hooks.hook_context.stage_execution_context') as mock_context:
+            # Mock the context manager to set execution_id on the model
+            async def mock_aenter(self):
```

Ruff (0.14.8)

Unused function argument: `self`

(ARG001)